### PR TITLE
feat(jtk): use cache for boards list and sprints list commands

### LINKED ISSUE #277: jtk: use agile caches for board and sprint read commands
Parent: #274

## North-star excerpt

Agile cache: `jtk boards list`, `jtk sprints list`, and possibly `jtk sprints current` with clear freshness rules. Prefer cache-first reads when the relevant envelope exists and is fresh; preserve live calls where cached data cannot preserve pagination, filtering, or freshness-sensitive behavior.

## Scope

- Evaluate and implement cache-first `jtk boards list` for cases that can be represented from the `boards` cache.
- Evaluate and implement cache-first `jtk sprints list --board` using the board-keyed `sprints` cache, including state filtering.
- Decide explicitly whether `jtk sprints current` should use cached active sprint data or remain live; document the reason in code/tests.
- Preserve board/project name resolution behavior and Agile unavailable behavior.

## Success criteria

- Fresh `boards` cache can satisfy supported `jtk boards list` invocations without a live `ListBoards` request.
- Fresh `sprints` cache can satisfy supported `jtk sprints list` invocations without a live `ListSprints` request.
- Unsupported cache cases, stale/missing/malformed cache, pagination cases that cannot be represented safely, and Agile-unavailable cases retain current behavior/errors.
- Tests cover project-filtered boards, board-scoped sprints, state filters, cache miss fallback, and any deliberate live-only decision for `sprints current`.
- Existing text, JSON, `--id`, `--fields`, `--extended`, and pagination contracts are preserved.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Route `jtk boards list` through the `boards` cache with client-side project filtering and pagination when cache is fresh
- Route `jtk sprints list` through the `sprints` cache with local state filtering; comma-separated state filters bypass cache for Jira server-side semantics
- `jtk sprints current` remains deliberately live — the active sprint is freshness-sensitive and cached data may lag sprint transitions
- Stale/missing/error cache falls back to existing live API paths; no write-through on fallback

## Test plan
- [x] Boards cache helper: fresh hit, project filter, pagination, project+pagination, manual TTL, stale/miss/uninitialized/no-instance fallback, live error propagation (10 tests)
- [x] Sprints cache helper: fresh hit, state filter, comma-state fallback, missing board fallback, empty board, manual TTL, stale/miss/no-instance fallback, live error propagation (11 tests)
- [x] Command-level: boards list and sprints list with fresh cache skip live (table + ID-only)
- [x] Command-level: sprints current always calls live API (deliberate live-only test)
- [x] Full suite: 1760 tests pass, lint clean

Closes #277

### CODE REVIEW

**Comment by rianjs** [comment]
## Findings

No remaining architectural or product-adherence findings.

The updated `sprints` helper now keeps cache use to the safe cases: empty state or known single states only, with comma-separated and unknown state values falling back live so Jira keeps its server-side semantics. `sprints current` remains live-only, which matches the freshness-sensitive product intent. Boards list also preserves the cache-first pattern with fresh/manual cache use, stale/miss/error live fallback, client-side project filtering, pagination, and no write-through.

I could not independently rerun the Go tests in this sandbox because `httptest` port binding is blocked here; this review is by diff/code-path inspection.

**Comment by rianjs** [comment]
## TDD Coverage Assessment

Coverage is solid overall. Tests are behaviorally meaningful (not just smoke tests), and the fallback paths are well-exercised. A few gaps worth noting:

### `boardsFromCache` — uncovered pagination edge cases

The `boardsFromCache` function clamps `startAt` to `total` when it overshoots, but there's no test for:

- **`startAt == total`** (exact boundary): should return an empty `Values` slice with `IsLast: true`
- **`startAt > total` after filtering**: e.g. request `startAt=5` against a project-filtered result of 2 boards — the clamp path is untested with a filter applied
- **`maxResults <= 0`** default fallback: the `maxResults = 50` substitution is never exercised by a test

### `GetSprintsCacheFirst` — state filter case sensitivity

The `isCacheableState` function normalizes to lowercase before switching, and `EqualFold` is used when filtering. There's no test that passes a mixed-case cacheable state (e.g. `"Active"` or `"FUTURE"`) to verify the path from isCacheable check through the filter works end-to-end. The individual pieces are correct, but the round-trip isn't exercised.

### `sprints_test.go` — stale/miss fallback with `stubFetcher` inconsistency

`TestGetSprintsCacheFirst_StaleCacheFallsBackToLive` and `TestGetSprintsCacheFirst_MissFallsBackToLive` both set up an httptest server that calls `t.Fatal`, then use `stubFetcher` instead of a real HTTP client — so the httptest server is unreachable dead code in those tests. The tests pass correctly, but the server setup is misleading. Minor cleanup opportunity.

### Not missing: the important paths are covered

- Fresh cache skips live (both resources)
- Stale/uninitialized/miss all fall back to live
- Project filter and pagination logic tested independently and combined
- Manual TTL treated as fresh
- Missing instance falls back to live
- Live error propagation tested for both resources
- `sprints current` deliberately bypasses cache (and has a test asserting that)

**Comment by rianjs** [comment]
Addressing TDD findings:

1. **Pagination boundary cases** — added `TestGetBoardsCacheFirst_FreshCacheStartAtBeyondTotal` and `TestGetBoardsCacheFirst_FreshCacheMaxResultsZeroUsesDefault`.
2. **Mixed-case state round-trip** — added `TestGetSprintsCacheFirst_MixedCaseStateServedFromCache` which passes `"Active"` (mixed case) to verify both `isCacheableState` and `EqualFold` cooperate.
3. **Dead httptest servers in stub tests** — the servers are a safety net: if the stub fetcher fails to intercept, the httptest server catches the unintended live call. Leaving as-is.

File: tools/jtk/internal/cache/boards_lookup.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	if maxResults <= 0 {
+		maxResults = 50
+	}
+
+	total := len(filtered)
+	if startAt > total {
+		startAt = total
+	}
+	end := startAt + maxResults
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/boards_lookup.go:60]: 🟡 **Medium** (harness-engineering:harness-architecture-reviewer): boardsFromCache does not populate the Total field in the returned BoardsResponse. Live calls return the server-populated Total; cache hits silently return 0. Any caller that uses Total for display or pagination decisions will observe wrong data on cache hits. Confirm that runList and all callers rely exclusively on IsLast, not Total — or set Total: len(filtered) to keep the cache path behaviorally equivalent to live.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/boards_lookup.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+// client.ListBoards on stale/miss/error.
+//
+// The boards cache stores []api.Board (all boards, exhaustively paginated by
+// fetchBoards). Client-side filtering and slicing reproduce the server-side
+// pagination contract.
+func GetBoardsCacheFirst(ctx context.Context, client *api.Client, projectFilter string, startAt, maxResults int) (*api.BoardsResponse, error) {
+	entry, err := Lookup("boards")
+	if err != nil {
+		return client.ListBoards(ctx, projectFilter, startAt, maxResults)
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/boards_lookup.go:22]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): GetBoardsCacheFirst calls Lookup("boards") and then ReadResource[[]api.Board]("boards") as two separate filesystem reads. A concurrent cache write between these two reads could cause Classify to evaluate a mismatched TTL/envelope pair. This is low-probability in a CLI context but creates a TOCTOU window not present in a single-read pattern. The same double-read issue exists in sprints_lookup.go:29.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/boards_lookup.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	env, err := ReadResource[[]api.Board]("boards")
+	if err != nil {
+		return client.ListBoards(ctx, projectFilter, startAt, maxResults)
+	}
+
+	switch Classify(env.FetchedAt, entry.TTL, time.Now()) {
+	case StatusFresh, StatusManual:
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/boards_lookup.go:30]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The switch over Classify has StatusStale, StatusUninitialized, and StatusUnavailable as separate cases that all execute the same live fallback, followed by an unreachable return. A new Status value added later would silently hit the post-switch fallback instead of producing a compile error. Collapse the three redundant arms into a default case to remove the dead path and make the contract explicit. Same pattern in sprints_lookup.go:44.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/sprints_lookup.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	case StatusFresh, StatusManual:
+		sprints, ok := env.Data[boardID]
+		if !ok {
+			return fetcher(ctx, client, boardID, state)
+		}
+		if state == "" {
+			return sprints, nil
+		}
+		var filtered []api.Sprint
+		for _, s := range sprints {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/sprints_lookup.go:45]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): When state != "" and no sprints match the filter, the function returns a nil slice rather than []api.Sprint{}. The live fetcher path returns whatever the API returns (typically an empty slice). Callers that test result != nil or marshal the result will observe a behavioral difference from the live path. Initialize filtered as var filtered []api.Sprint{} or document the nil contract explicitly.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/sprints_lookup_test.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+		t.Fatal("expected live fetcher call for comma-separated state")
+	}
+	testutil.Equal(t, len(got), 2)
+}
+
+func TestGetSprintsCacheFirst_MixedCaseStateServedFromCache(t *testing.T) {
+	t.Cleanup(SetRootForTest(t.TempDir()))
+	t.Cleanup(SetInstanceKeyForTest("test.atlassian.net"))
+
+	testutil.RequireNoError(t, WriteResource("sprints", "24h", testSprints))
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/sprints_lookup_test.go:118]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): TestGetSprintsCacheFirst_FreshCacheEmptyBoardReturnsEmpty only covers state="" for an empty sprint list. There is no test for: board in cache with an empty sprint list and a state filter applied. In that path the filter loop runs over an empty slice and returns nil rather than []api.Sprint{}, which is a gap for the nil-vs-empty-slice contract noted in sprints_lookup.go.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 6]
```diff
@@ -1553,3 +1553,100 @@ func TestRunAdd_AgentOutput(t *testing.T) {
...
+	err = runList(context.Background(), opts, client, 1, "", 50, "", "")
+	testutil.RequireNoError(t, err)
+	testutil.Contains(t, stdout.String(), "100\n")
+	testutil.Contains(t, stdout.String(), "101\n")
+}
+
+// sprints current is deliberately live-only: the active sprint is
+// freshness-sensitive and may lag sprint transitions in cache.
+func TestRunCurrent_AlwaysCallsLive(t *testing.T) {
+	seedBoardsAndSprints(t,
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:1618]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): TestRunCurrent_AlwaysCallsLive seeds the sprints cache but does not assert that the cache is NOT consulted. The test verifies live is called, but a future refactor routing runCurrent through GetSprintsCacheFirst would still pass as long as live is also called. An explicit assertion that output matches live data (distinct from the seeded cached sprint) would make the test's intent more robust.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `e20786f`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 5 |
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (5 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/internal/cache/boards_lookup.go:60`
> boardsFromCache does not populate the Total field in the returned BoardsResponse. Live calls return the server-populated Total; cache hits silently return 0. Any caller that uses Total for display or pagination decisions will observe wrong data on cache hits. Confirm that runList and all callers rely exclusively on IsLast, not Total — or set Total: len(filtered) to keep the cache path behaviorally equivalent to live.

**💡 Suggestion** - `tools/jtk/internal/cache/boards_lookup.go:22`
> GetBoardsCacheFirst calls Lookup("boards") and then ReadResource[[]api.Board]("boards") as two separate filesystem reads. A concurrent cache write between these two reads could cause Classify to evaluate a mismatched TTL/envelope pair. This is low-probability in a CLI context but creates a TOCTOU window not present in a single-read pattern. The same double-read issue exists in sprints_lookup.go:29.

**💡 Suggestion** - `tools/jtk/internal/cache/boards_lookup.go:30`
> The switch over Classify has StatusStale, StatusUninitialized, and StatusUnavailable as separate cases that all execute the same live fallback, followed by an unreachable return. A new Status value added later would silently hit the post-switch fallback instead of producing a compile error. Collapse the three redundant arms into a default case to remove the dead path and make the contract explicit. Same pattern in sprints_lookup.go:44.

**💡 Suggestion** - `tools/jtk/internal/cache/sprints_lookup.go:45`
> When state != "" and no sprints match the filter, the function returns a nil slice rather than []api.Sprint{}. The live fetcher path returns whatever the API returns (typically an empty slice). Callers that test result != nil or marshal the result will observe a behavioral difference from the live path. Initialize filtered as var filtered []api.Sprint{} or document the nil contract explicitly.

**💡 Suggestion** - `tools/jtk/internal/cache/sprints_lookup_test.go:118`
> TestGetSprintsCacheFirst_FreshCacheEmptyBoardReturnsEmpty only covers state="" for an empty sprint list. There is no test for: board in cache with an empty sprint list and a state filter applied. In that path the filter loop runs over an empty slice and returns nil rather than []api.Sprint{}, which is a gap for the nil-vs-empty-slice contract noted in sprints_lookup.go.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1618`
> TestRunCurrent_AlwaysCallsLive seeds the sprints cache but does not assert that the cache is NOT consulted. The test verifies live is called, but a future refactor routing runCurrent through GetSprintsCacheFirst would still pass as long as live is also called. An explicit assertion that output matches live data (distinct from the seeded cached sprint) would make the test's intent more robust.

</details>

*3 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 43s | $1.07</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 43s wall · 7m 21s compute (Reviewers: 1m 24s · Synthesis: 1m 16s) |
| Cost | $1.07 |
| Tokens | 511.8k in / 29.4k out |
| Turns | 22 |

</details>

File: tools/jtk/internal/cache/boards_lookup.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/internal/cache/boards_lookup.go:60]: Fixed — `boardsFromCache` now sets `Total: len(filtered)`. `runList` only uses `IsLast` and `Values`, but this keeps the cache path behaviorally equivalent to live.

File: tools/jtk/internal/cache/boards_lookup.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cache/boards_lookup.go:22]: Accepted risk — this is a single-process CLI and the window between Lookup and ReadResource is microseconds. The same pattern exists in all `GetXxxCacheFirst` helpers from #275 and #276. Not worth adding complexity for a non-concurrent scenario.

File: tools/jtk/internal/cache/boards_lookup.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/cache/boards_lookup.go:30]: Intentional — matches the established pattern in `fields_lookup.go` (lines 37–48) which keeps all Status constants explicit. The trailing return satisfies the compiler. Consistency across all lookup helpers is more important than collapsing here.

File: tools/jtk/internal/cache/sprints_lookup.go (hunk 4 above)
Comment (rianjs) [reply tools/jtk/internal/cache/sprints_lookup.go:45]: Callers (`runList`) handle both nil and empty identically — `len(nil) == 0` and `append(nil, ...)...` both work. No caller tests `result != nil`. JSON marshaling of nil slice vs empty slice is not relevant here since the command handles presentation after the cache helper returns.

File: tools/jtk/internal/cache/sprints_lookup_test.go (hunk 5 above)
Comment (rianjs) [reply tools/jtk/internal/cache/sprints_lookup_test.go:118]: The nil-vs-empty distinction is not observable by callers (see reply above). The empty board test covers the key behavior: board present in cache → no live fallback. State filter over empty is a subset of that.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 6 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1618]: The test's intent is to verify the live API is called — if a future refactor routes `runCurrent` through cache, the test should still pass because `runCurrent` must call live regardless. The cached and live data being identical is fine because the assertion is about the call path, not the data.

File: tools/jtk/internal/cache/boards_lookup.go (hunk 1 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cache/boards_lookup.go:60]: **Settled**: monit-reviewer flagged that boardsFromCache did not populate the Total field, causing cache hits to silently return 0. rianjs fixed it by setting Total: len(filtered) to keep the cache path behaviorally equivalent to live.

File: tools/jtk/internal/cache/boards_lookup.go (hunk 2 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cache/boards_lookup.go:22]: **Settled**: monit-reviewer noted a TOCTOU window from two separate filesystem reads in GetBoardsCacheFirst and GetSprintsCacheFirst. rianjs accepted the risk, citing the single-process CLI context and consistency with the same pattern established in prior PRs.

File: tools/jtk/internal/cache/boards_lookup.go (hunk 3 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cache/boards_lookup.go:30]: **Settled**: monit-reviewer suggested collapsing redundant switch cases into a default to avoid silent fallback on new Status values. rianjs declined, noting this matches the explicit pattern in fields_lookup.go and that consistency across lookup helpers takes priority.

File: tools/jtk/internal/cache/sprints_lookup.go (hunk 4 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cache/sprints_lookup.go:45]: **Settled**: monit-reviewer flagged that filtered sprints returns nil rather than an empty slice when no sprints match the state filter. rianjs explained that callers handle nil and empty identically and no caller tests result != nil, so the distinction is not observable.

File: tools/jtk/internal/cache/sprints_lookup_test.go (hunk 5 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cache/sprints_lookup_test.go:118]: **Settled**: monit-reviewer suggested adding a test for state filter over an empty sprint list to cover the nil-vs-empty contract. rianjs noted the nil-vs-empty distinction is not observable by callers and the existing empty board test covers the key behavior of no live fallback.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 6 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1618]: **Settled**: monit-reviewer suggested adding an assertion that cache is not consulted in TestRunCurrent_AlwaysCallsLive to make the test more robust. rianjs explained the test's intent is to verify the live API is called, and a future refactor routing through cache should still pass as long as live is also called.

File: tools/jtk/internal/cache/boards_lookup.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	if projectFilter != "" {
+		for _, b := range boards {
+			if strings.EqualFold(b.Location.ProjectKey, projectFilter) {
+				filtered = append(filtered, b)
+			}
+		}
+	} else {
+		filtered = boards
+	}
+
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/boards_lookup.go:51]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): When maxResults <= 0, boardsFromCache silently defaults to 50, but the live path forwards 0 directly to client.ListBoards. If the Jira API interprets maxResults=0 as 'no limit' or its own server default, the cache and live paths return different page sizes for the same inputs, breaking pagination consistency.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/boards_lookup.go [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+		maxResults = 50
+	}
+
+	total := len(filtered)
+	if startAt > total {
+		startAt = total
+	}
+	end := startAt + maxResults
+	if end > total {
+		end = total
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/boards_lookup.go:62]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): When startAt > total, boardsFromCache clamps it to total before building the response, so BoardsResponse.StartAt reflects the clamped value rather than the originally requested offset. The live Jira API echoes back the requested startAt regardless of whether it exceeds the total. Callers that use the response's StartAt field to detect over-pagination will see divergent behavior between cache and live paths.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/boards_lookup.go [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+// GetBoardsCacheFirst returns a BoardsResponse from the boards cache when
+// fresh, applying project filter and pagination client-side. Falls back to
+// client.ListBoards on stale/miss/error.
+//
+// The boards cache stores []api.Board (all boards, exhaustively paginated by
+// fetchBoards). Client-side filtering and slicing reproduce the server-side
+// pagination contract.
+func GetBoardsCacheFirst(ctx context.Context, client *api.Client, projectFilter string, startAt, maxResults int) (*api.BoardsResponse, error) {
+	entry, err := Lookup("boards")
+	if err != nil {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/boards_lookup.go:20]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): GetBoardsCacheFirst hardcodes client.ListBoards as the live fallback while GetSprintsCacheFirst accepts an injectable fetcher function. The boards command tests compensate by spinning up a real httptest.NewServer, making them heavier and more coupled to HTTP behavior. Aligning the design would make both helpers equally testable without a running HTTP server.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `5a1d8c1` | **Previous:** `e20786f` (incremental)

> **Approved** with 3 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-knowledge-reviewer | 2 |
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cache/boards_lookup.go:62`
> When startAt > total, boardsFromCache clamps it to total before building the response, so BoardsResponse.StartAt reflects the clamped value rather than the originally requested offset. The live Jira API echoes back the requested startAt regardless of whether it exceeds the total. Callers that use the response's StartAt field to detect over-pagination will see divergent behavior between cache and live paths.

**💡 Suggestion** - `tools/jtk/internal/cache/boards_lookup.go:20`
> GetBoardsCacheFirst hardcodes client.ListBoards as the live fallback while GetSprintsCacheFirst accepts an injectable fetcher function. The boards command tests compensate by spinning up a real httptest.NewServer, making them heavier and more coupled to HTTP behavior. Aligning the design would make both helpers equally testable without a running HTTP server.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cache/boards_lookup.go:51`
> When maxResults <= 0, boardsFromCache silently defaults to 50, but the live path forwards 0 directly to client.ListBoards. If the Jira API interprets maxResults=0 as 'no limit' or its own server default, the cache and live paths return different page sizes for the same inputs, breaking pagination consistency.

</details>

*9 PR discussion threads considered.*

---
<details>
<summary>Completed in 3m 18s | $1.20</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 3m 18s wall · 9m 18s compute (Reviewers: 2m 27s · Synthesis: 29s) |
| Cost | $1.20 |
| Tokens | 408.1k in / 34.8k out |
| Turns | 18 |

</details>
